<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with scIso • scIso</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with scIso">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scIso</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting started with scIso</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/alihamraoui/scIso/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Getting started with scIso</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/alihamraoui/scIso/blob/master/vignettes/getting-started.Rmd" class="external-link"><code>vignettes/getting-started.Rmd</code></a></small>
      <div class="d-none name"><code>getting-started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">1. Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>scIso provides tools to analyse isoform usage in single-cell
long-read RNA-seq. Starting from an isoform count matrix and a standard
Seurat object, scIso:</p>
<ul>
<li><p>computes isoform-level metrics (Percent Inclusion, PI;
heterogeneity, PHI),</p></li>
<li><p>visualises isoform usage on embeddings,</p></li>
<li><p>links isoform dynamics to pseudotime.</p></li>
</ul>
<p>This vignette demonstrates a minimal end-to-end workflow using an
example MPNST dataset downloaded from Zenodo.</p>
<blockquote>
<h3 id="workflow-summary">Workflow summary</h3>
<ul>
<li><p>Load isoform counts and metadata (cell-type annotation,
pseudotime).</p></li>
<li><p>Build a Seurat object and perform standard
preprocessing.</p></li>
<li><p>Visualise clusters and cell-type annotation.</p></li>
<li><p>Compute and explore PI/PHI from an isoform count matrix.</p></li>
<li><p>Project isoform usage on embeddings and along
pseudotime.</p></li>
<li><p>Briefly indicate how to adapt the workflow to your own
data.</p></li>
</ul>
</blockquote>
</div>
<div class="section level2">
<h2 id="load-packages">2. Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<p>We load scIso together with the core dependencies used in this
vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alihamraoui.github.io/scIso">scIso</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-example-data-from-zenodo">3. Load example data from Zenodo<a class="anchor" aria-label="anchor" href="#load-example-data-from-zenodo"></a>
</h2>
<p>We use a pre-computed cell-type annotation, a pseudotime vector and
an isoform count matrix hosted on Zenodo.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/gzcon.html" class="external-link">gzcon</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/17585277/files/Cell_type_annotation.rds"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pseudotime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/gzcon.html" class="external-link">gzcon</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/17654917/files/Pseudotime_2020_23.rds"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu">curl</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/curl/reference/nslookup.html" class="external-link">has_internet</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".zip"</span><span class="op">)</span></span>
<span>  <span class="fu">curl</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/curl/reference/curl_download.html" class="external-link">curl_download</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/17585277/files/iso_matrix.zip?download=1"</span>, <span class="va">tmp</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="va">tmp</span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">scmat_iso</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-a-seurat-object-and-run-standard-preprocessing">4. Build a Seurat object and run standard preprocessing<a class="anchor" aria-label="anchor" href="#build-a-seurat-object-and-run-standard-preprocessing"></a>
</h2>
<p>We construct a Seurat object using the isoform counts as a dedicated
assay and apply a standard preprocessing pipeline (normalisation,
variable feature selection, PCA, UMAP/t-SNE, neighbour graph,
clustering).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assay</span> <span class="op">=</span> <span class="st">"ISO"</span></span>
<span><span class="va">iso_assay</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">scmat_iso</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">iso_assay</span>,</span>
<span>                                        assay <span class="op">=</span> <span class="va">assay</span>,</span>
<span>                                        project.name <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'log_nCount_'</span>, <span class="va">assay</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'nCount_'</span>, <span class="va">assay</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 93670 features across 4532 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: ISO (93670 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  2 layers present: counts, data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">ISO</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "BambuGene6100..BambuTx23926" "BambuGene1971..BambuTx30365"</span></span>
<span><span class="co">#&gt; [3] "BambuGene3230..BambuTx42042" "BambuGene8892..BambuTx58725"</span></span>
<span><span class="co">#&gt; [5] "Mup10..BambuTx68486"         "Mup12..BambuTx68510"</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                                   normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>                                   assay <span class="op">=</span> <span class="st">"ISO"</span>,</span>
<span>                                   verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                                          assay <span class="op">=</span> <span class="st">"ISO"</span>,</span>
<span>                                          nfeatures <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>                                          verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 93670 features across 4532 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: ISO (93670 features, 3000 variable features)</span></span>
<span><span class="co">#&gt;  2 layers present: counts, data</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">common_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">[</span>,<span class="va">common_cells</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">=</span> <span class="st">"ISO"</span></span>
<span><span class="va">ndims</span> <span class="op">=</span> <span class="fl">18</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                               features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, </span>
<span>                               verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="va">var_features</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>                            features <span class="op">=</span> <span class="va">var_features</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                            reduction.name <span class="op">=</span> <span class="st">"RNA_pca"</span>, </span>
<span>                            max_dims <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                             reduction <span class="op">=</span> <span class="st">"RNA_pca"</span>,</span>
<span>                             dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">ndims</span>,</span>
<span>                             seed.use <span class="op">=</span> <span class="fl">1337L</span>,</span>
<span>                             verbose <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                             reduction.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_"</span>, <span class="va">ndims</span>, <span class="st">"_umap"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span></span>
<span><span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span></span>
<span><span class="co">#&gt; This message will be shown once per session</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                             reduction <span class="op">=</span> <span class="st">"RNA_pca"</span>,</span>
<span>                             dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">ndims</span>,</span>
<span>                             seed.use <span class="op">=</span> <span class="fl">1337L</span>,</span>
<span>                             verbose <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                             reduction.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_"</span>, <span class="va">ndims</span>, <span class="st">"_tsne"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 93670 features across 3566 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: ISO (93670 features, 3000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  3 dimensional reductions calculated: RNA_pca, RNA_pca_18_umap, RNA_pca_18_tsne</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">seurat_obj</span>, reduction <span class="op">=</span> <span class="st">"RNA_pca"</span>, dims <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">ndims</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Computing nearest neighbor graph</span></span>
<span><span class="co">#&gt; Computing SNN</span></span>
<span><span class="va">seurat_obj</span> <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seurat_obj</span>, resolution <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 3566</span></span>
<span><span class="co">#&gt; Number of edges: 128336</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.8735</span></span>
<span><span class="co">#&gt; Number of communities: 9</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="add-cell-type-annotation-and-visualise-clusters">5. Add cell-type annotation and visualise clusters<a class="anchor" aria-label="anchor" href="#add-cell-type-annotation-and-visualise-clusters"></a>
</h2>
<p>We now bring in the external cell-type labels and overlay them on the
embedding.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_types_named</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span><span class="op">[</span><span class="va">cell_types</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cell_types_named</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="cn">NA_character_</span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">cell_types</span><span class="op">[</span><span class="va">common_cells</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">cell_types_named</span><span class="op">[</span><span class="va">common_cells</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="st">"tsne"</span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>                      <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">proj</span>,<span class="st">"\n"</span>, <span class="st">"MPNST"</span><span class="op">)</span>,</span>
<span>                                    subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="st">" cells"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                      <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>                      <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                     plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>                                     plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">|</span></span>
<span></span>
<span>            <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span>, group.by <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">proj</span>,<span class="st">"\n"</span>, <span class="st">"MPNST"</span><span class="op">)</span>,</span>
<span>                                  subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="st">" cells"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                   plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>                                   plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/cell_viz-1.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#p_clusters | p_annot</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="compute-and-explore-isoform-piphi">6. Compute and explore isoform PI/PHI<a class="anchor" aria-label="anchor" href="#compute-and-explore-isoform-piphi"></a>
</h2>
<p>scIso uses PI (Percent Inclusion) and PHI (heterogeneity) to
summarise isoform usage per gene. For speed, we load a pre-computed PI
matrix, but the same result can be obtained with computePIMatrix() on
the fly.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#pi_mtx &lt;- scIso::computePIMatrix(seurat_obj, </span></span>
<span><span class="co">#                                 assay = "ISO",</span></span>
<span><span class="co">#                                 slot = "count")</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu">curl</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/curl/reference/nslookup.html" class="external-link">has_internet</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".rds"</span><span class="op">)</span></span>
<span>  <span class="fu">curl</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/curl/reference/curl_download.html" class="external-link">curl_download</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/17585277/files/pi_mtx.rds?download=1"</span>, <span class="va">tmp</span><span class="op">)</span></span>
<span>  <span class="va">pi_mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scIso</span><span class="fu">::</span><span class="fu"><a href="../reference/plotPIPHI.html">plotPIPHI</a></span><span class="op">(</span>pi_mat <span class="op">=</span> <span class="va">pi_mtx</span>,</span>
<span>                 iso_mat <span class="op">=</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">ISO</span><span class="op">$</span><span class="va">counts</span>,</span>
<span>                 min_gene_count <span class="op">=</span> <span class="fl">3000</span></span>
<span>                 <span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/plot_pi-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level2">
<h2 id="visualise-isoform-usage-on-embeddings">7. Visualise isoform usage on embeddings<a class="anchor" aria-label="anchor" href="#visualise-isoform-usage-on-embeddings"></a>
</h2>
<p>We next contrast two isoforms of Srsf3 on the t-SNE embedding, using
FeaturePlotPI(). This function computes PI for the selected isoforms and
displays PI and log-ratio at the single-cell level.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">scIso</span><span class="fu">::</span><span class="fu"><a href="../reference/FeaturePlotPI.html">FeaturePlotPI</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>                     name <span class="op">=</span> <span class="st">"Srsf3"</span>, </span>
<span>                     reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span>,</span>
<span>                     iso1 <span class="op">=</span> <span class="st">"Srsf3..ENSMUST00000130216"</span>,</span>
<span>                     iso2 <span class="op">=</span> <span class="st">"Srsf3..ENSMUST00000037776"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `GetAssayData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">scIso</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/alihamraoui/scIso/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="va">p</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">|</span> <span class="va">p</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/plot_ration_and_pi_dist-1.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">|</span> <span class="va">p</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/plot_iso-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level2">
<h2 id="restrict-to-tumour-cells-and-add-pseudotime">8. Restrict to tumour cells and add pseudotime<a class="anchor" aria-label="anchor" href="#restrict-to-tumour-cells-and-add-pseudotime"></a>
</h2>
<p>We focus on tumour cells, refine the annotation, and add the
pseudotime trajectory previously inferred for this dataset.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">major</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">seurat_clusters</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>cell_annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">seurat_clusters</span>, <span class="va">major</span><span class="op">$</span><span class="va">seurat_clusters</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">cell_pop</span> <span class="op">&lt;-</span> <span class="va">major</span><span class="op">$</span><span class="va">cell_annotation</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seurat_obj</span>, subset <span class="op">=</span> <span class="va">cell_pop</span> <span class="op">==</span> <span class="st">"tumor cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">tumor_cell_type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op">==</span> <span class="fl">4</span>, <span class="st">"Sox10+"</span>, <span class="st">"Sox9+"</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">tumor_cell_type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op">==</span> <span class="fl">4</span>, <span class="st">"Sox10+"</span>, <span class="st">"Sox9+"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pseudotime</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".*_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pseudotime</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">pseudotime</span> <span class="op">&lt;-</span> <span class="va">pseudotime</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span>, group.by <span class="op">=</span> <span class="st">"cell_pop"</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">proj</span>,<span class="st">"\n"</span>, <span class="st">"MPNST"</span><span class="op">)</span>,</span>
<span>                                  subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="st">" cells"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                   plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>                                   plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">|</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span>, group.by <span class="op">=</span> <span class="st">"tumor_cell_type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">proj</span>,<span class="st">"\n"</span>, <span class="st">"MPNST"</span><span class="op">)</span>,</span>
<span>                                  subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="st">" cells"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>                    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                   plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>                                   plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">|</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features <span class="op">=</span> <span class="st">"pseudotime"</span>,</span>
<span>                    reduction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RNA_pca_18_"</span>, <span class="va">proj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Spectral"</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Pseudotime"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/viz_cells_annotations-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level2">
<h2 id="isoform-dynamics-along-pseudotime">9. Isoform dynamics along pseudotime<a class="anchor" aria-label="anchor" href="#isoform-dynamics-along-pseudotime"></a>
</h2>
<p>IsoformPseudotimePlot() summarises isoform usage for a given gene
across pseudotime bins and optionally stratified by cell groups.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scIso</span><span class="fu">::</span><span class="fu"><a href="../reference/IsoformPseudotimePlot.html">IsoformPseudotimePlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>                             gene <span class="op">=</span> <span class="st">"Dnaja1"</span>,</span>
<span>                             band_cols <span class="op">=</span> <span class="st">"tumor_cell_type"</span>,</span>
<span>                             pseudotime_col <span class="op">=</span> <span class="st">"pseudotime"</span>,</span>
<span>                             nbins <span class="op">=</span> <span class="fl">80</span>,</span>
<span>                             mode <span class="op">=</span> <span class="st">"PI"</span>,</span>
<span>                             slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">scIso</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/alihamraoui/scIso/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; `geom_smooth()` using formula = 'y ~ x'</span></span>
<span><span class="co">#&gt; Warning: Removed 764 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_smooth()`).</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/plot_pstime-1.png" class="r-plt" width="2496"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scIso</span><span class="fu">::</span><span class="fu"><a href="../reference/IsoformPseudotimePlot.html">IsoformPseudotimePlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>                             gene <span class="op">=</span> <span class="st">"Dnaja1"</span>,</span>
<span>                             band_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tumor_cell_type"</span>, <span class="st">"tumor_cell_type"</span><span class="op">)</span>,</span>
<span>                             pseudotime_col <span class="op">=</span> <span class="st">"pseudotime"</span>, </span>
<span>                             nbins <span class="op">=</span> <span class="fl">30</span>,</span>
<span>                             mode <span class="op">=</span> <span class="st">"exp"</span>,</span>
<span>                             slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="co">#&gt; `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/unnamed-chunk-2-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level2">
<h2 id="using-sciso-on-your-own-data">10. Using scIso on your own data<a class="anchor" aria-label="anchor" href="#using-sciso-on-your-own-data"></a>
</h2>
<ul>
<li><p>To apply this workflow to your own dataset:</p></li>
<li><p>replace the example isoform matrix with your own (features =
isoforms or isoform-level identifiers, columns =
cells/barcodes);</p></li>
<li><p>construct a Seurat object with an isoform assay (e.g. “ISO”) and
follow your preferred preprocessing strategy;</p></li>
<li><p>compute PI using computePIMatrix() on your isoform count
matrix;</p></li>
<li><p>add any relevant metadata (cell types, clusters, trajectories) to
<a href="mailto:seurat_obj@meta.data" class="email">seurat_obj@meta.data</a>;</p></li>
<li><p>use plotPIPHI(), FeaturePlotPI() and IsoformPseudotimePlot() to
interrogate isoform heterogeneity and dynamics.</p></li>
</ul>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] future_1.68.0 dplyr_1.1.4   scIso_0.1.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3     jsonlite_2.0.0         magrittr_2.0.4        </span></span>
<span><span class="co">#&gt;   [4] spatstat.utils_3.2-0   farver_2.1.2           rmarkdown_2.30        </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.6               ragg_1.5.0             vctrs_0.6.5           </span></span>
<span><span class="co">#&gt;  [10] ROCR_1.0-11            spatstat.explore_3.5-3 paletteer_1.6.0       </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.8.1      curl_7.0.0             sass_0.4.10           </span></span>
<span><span class="co">#&gt;  [16] sctransform_0.4.2      parallelly_1.45.1      KernSmooth_2.23-26    </span></span>
<span><span class="co">#&gt;  [19] bslib_0.9.0            htmlwidgets_1.6.4      desc_1.4.3            </span></span>
<span><span class="co">#&gt;  [22] ica_1.0-3              plyr_1.8.9             plotly_4.11.0         </span></span>
<span><span class="co">#&gt;  [25] zoo_1.8-14             cachem_1.1.0           igraph_2.2.1          </span></span>
<span><span class="co">#&gt;  [28] mime_0.13              lifecycle_1.0.4        pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.7-4           R6_2.6.1               fastmap_1.2.0         </span></span>
<span><span class="co">#&gt;  [34] fitdistrplus_1.2-4     shiny_1.11.1           digest_0.6.39         </span></span>
<span><span class="co">#&gt;  [37] rematch2_2.1.2         patchwork_1.3.2        Seurat_5.3.1          </span></span>
<span><span class="co">#&gt;  [40] tensor_1.5.1           prismatic_1.1.2        RSpectra_0.16-2       </span></span>
<span><span class="co">#&gt;  [43] irlba_2.3.5.1          textshaping_1.0.4      labeling_0.4.3        </span></span>
<span><span class="co">#&gt;  [46] progressr_0.18.0       latex2exp_0.9.6        spatstat.sparse_3.1-0 </span></span>
<span><span class="co">#&gt;  [49] mgcv_1.9-3             httr_1.4.7             polyclip_1.10-7       </span></span>
<span><span class="co">#&gt;  [52] abind_1.4-8            compiler_4.5.2         withr_3.0.2           </span></span>
<span><span class="co">#&gt;  [55] S7_0.2.1               fastDummies_1.7.5      MASS_7.3-65           </span></span>
<span><span class="co">#&gt;  [58] tools_4.5.2            lmtest_0.9-40          otel_0.2.0            </span></span>
<span><span class="co">#&gt;  [61] httpuv_1.6.16          future.apply_1.20.0    goftest_1.2-3         </span></span>
<span><span class="co">#&gt;  [64] glue_1.8.0             nlme_3.1-168           promises_1.5.0        </span></span>
<span><span class="co">#&gt;  [67] grid_4.5.2             Rtsne_0.17             cluster_2.1.8.1       </span></span>
<span><span class="co">#&gt;  [70] reshape2_1.4.5         generics_0.1.4         gtable_0.3.6          </span></span>
<span><span class="co">#&gt;  [73] spatstat.data_3.1-9    tidyr_1.3.1            data.table_1.17.8     </span></span>
<span><span class="co">#&gt;  [76] sp_2.2-0               spatstat.geom_3.6-1    RcppAnnoy_0.0.22      </span></span>
<span><span class="co">#&gt;  [79] ggrepel_0.9.6          RANN_2.6.2             pillar_1.11.1         </span></span>
<span><span class="co">#&gt;  [82] stringr_1.6.0          spam_2.11-1            RcppHNSW_0.6.0        </span></span>
<span><span class="co">#&gt;  [85] later_1.4.4            splines_4.5.2          lattice_0.22-7        </span></span>
<span><span class="co">#&gt;  [88] survival_3.8-3         deldir_2.0-4           tidyselect_1.2.1      </span></span>
<span><span class="co">#&gt;  [91] miniUI_0.1.2           pbapply_1.7-4          knitr_1.50            </span></span>
<span><span class="co">#&gt;  [94] gridExtra_2.3          scattermore_1.2        xfun_0.54             </span></span>
<span><span class="co">#&gt;  [97] matrixStats_1.5.0      stringi_1.8.7          lazyeval_0.2.2        </span></span>
<span><span class="co">#&gt; [100] yaml_2.3.10            evaluate_1.0.5         codetools_0.2-20      </span></span>
<span><span class="co">#&gt; [103] tibble_3.3.0           cli_3.6.5              uwot_0.2.4            </span></span>
<span><span class="co">#&gt; [106] xtable_1.8-4           reticulate_1.44.1      systemfonts_1.3.1     </span></span>
<span><span class="co">#&gt; [109] jquerylib_0.1.4        Rcpp_1.1.0             globals_0.18.0        </span></span>
<span><span class="co">#&gt; [112] spatstat.random_3.4-2  png_0.1-8              spatstat.univar_3.1-5 </span></span>
<span><span class="co">#&gt; [115] parallel_4.5.2         pkgdown_2.2.0          ggplot2_4.0.1         </span></span>
<span><span class="co">#&gt; [118] dotCall64_1.2          listenv_0.10.0         viridisLite_0.4.2     </span></span>
<span><span class="co">#&gt; [121] ggthemes_5.1.0         scales_1.4.0           ggridges_0.5.7        </span></span>
<span><span class="co">#&gt; [124] crayon_1.5.3           SeuratObject_5.2.0     purrr_1.2.0           </span></span>
<span><span class="co">#&gt; [127] rlang_1.1.6            cowplot_1.2.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ali Hamraoui.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
